### ISSUE TITLE
Code comment shortcut keys are invalid

### ISSUE THREAD

**Issue by Sun-Wen-Jie** [pulsar-edit/pulsar#732]
### Thanks in advance for your bug report!

- [X] Have you reproduced issue in [safe mode](https://pulsar-edit.dev/docs/atom-archive/faq/sections/what-does-safe-mode-do.html)?
- [X] Have you used the [debugging guide](https://pulsar-edit.dev/docs/atom-archive/hacking-atom/#debugging) to try to resolve the issue?
- [X] Have you checked our [FAQs](https://pulsar-edit.dev/docs/launch-manual/sections/faq/) to make sure your question isn't answered there?
- [X] Have you checked to make sure your issue does not already [exist](https://github.com/issues?q=is%3Aissue+user%3Apulsar-edit+is%3Aopen)?
- [X] Have you checked you are on the latest release of Pulsar?

### What happened?

When I downloaded and used pulsar to open the json file on my mac and wanted to comment part of the code, I found that commond + / could not be used. Clicking the switch comment in editing also did not work. I don’t know if there is a Chinese simplified Chinese plug-in that I downloaded. 

### Pulsar version

pulsar 1.109.0

### Which OS does this happen on?

🍎 macOS

### OS details

MacBookPro13.5.2 1.4 GHz 四核Intel Core i5

### Which CPU architecture are you running this on?

Apple M1/M2

### What steps are needed to reproduce this?

Install pulsar, then download the Chinese version plug-in and try to use the comment shortcut key commond + / or click the switch comment button in editing

### Additional Information:

![image](https://github.com/pulsar-edit/pulsar/assets/104260055/ea205fcc-2290-42e3-acc1-341d38773293)

**Comment by Sun-Wen-Jie** [comment]
Oh, this is not a bug, but because pulsar does not support comments on json files. I hope this function can be added in the future. Thanks.

**Comment by Daeraxa** [comment]
I think we should probably consider this as an enhancement to add "json with comments" support as an extension to 	our `language-json` grammar as a new scope. 

It seems this is exactly what VSCode have done - https://code.visualstudio.com/Docs/languages/json#_json-with-comments - in order to make sure that "vanilla" json can't be broken by making it seem like putting comments in is allowed, but "extended" json like json5 can have them added.

I think that if we were to do this the default for `.json` files would still have to be the "vanilla" grammar. What I don't know is if we could load the default grammar, check for comment syntax and then swap to the hypothetical new "json-c" grammar automatically.

**Comment by Sun-Wen-Jie** [comment]
> 我认为我们应该将其视为增强功能，添加“带注释的 json”支持作为我们`language-json`语法的扩展作为新范围。
> 
> 看起来这正是 VSCode 所做的 - https://code.visualstudio.com/Docs/languages/json#_json-with-comments - 为了确保“vanilla”json 不会被破坏似乎允许添加注释，但是像 json5 这样的“扩展”json 可以添加它们。
> 
> 我认为如果我们要这样做，`.json`文件的默认值仍然必须是“vanilla”语法。我不知道的是我们是否可以加载默认语法，检查注释语法，然后自动切换到假设的新“json-c”语法。

yes I think so,I hope to see this enhancement in subsequent versions, thanks for the reply！
